### PR TITLE
Add forgotten explicit nullable: true attribributes

### PR DESCRIPTION
**Pull request by edeutsch**

### CODE REVIEW

File: TranslatorReasonerAPI.yaml [hunk 1]
```diff
@@ -879,6 +886,7 @@ components:
             Node attributes provided by this TRAPI web service.
           items:
             $ref: '#/components/schemas/MetaAttribute'
+          nullable: true
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:889]: I would prefer a sensible default to `null`:
```suggestion
          default: []
```

File: TranslatorReasonerAPI.yaml [hunk 2]
```diff
@@ -909,6 +917,7 @@ components:
             Edge attributes provided by this TRAPI web service.
           items:
             $ref: '#/components/schemas/MetaAttribute'
+          nullable: true
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:920]:
```suggestion
          default: []
```

File: TranslatorReasonerAPI.yaml [hunk 3]
```diff
@@ -928,12 +937,14 @@ components:
...
             Source of an attribute provided by this TRAPI web service.
           example: infores:chembl
+          nullable: true
         original_attribute_names:
           type: array
           items:
             type: string
           description: >-
             Names of an the attribute as provided by the source.
+          nullable: true
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:947]:
```suggestion
          default: []
```

File: TranslatorReasonerAPI.yaml [hunk 4]
```diff
@@ -997,6 +1009,7 @@ components:
             - '>'
             - <
             - matches
+          nullable: false
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:1012]: `operator` is required.
```suggestion
```

File: TranslatorReasonerAPI.yaml [hunk 5]
```diff
@@ -1007,13 +1020,15 @@ components:
             A 'list' data type paired with the '>' or '<' operators will
             encode extraneous comparisons, but this is permitted as it is in
             SQL and other languages.
+          nullable: false
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:1023]:
```suggestion
```

**Review by patrickkwang — CHANGES_REQUESTED** [review]

**Review by kshefchek — APPROVED** [review]

File: TranslatorReasonerAPI.yaml (hunk 1 above)
Comment (patrickkwang) [reply TranslatorReasonerAPI.yaml:889]: Per WG discussion, the empty array here actually means a different thing than null.

File: TranslatorReasonerAPI.yaml (hunk 3 above)
Comment (patrickkwang) [reply TranslatorReasonerAPI.yaml:947]: Per WG discussion, we prefer null to empty array here. Instead, let's add `minItems: 1`.

**Review by marcdubybroad — APPROVED** [review]

**Review by kshefchek — APPROVED** [review]

**Review by vdancik — APPROVED** [review]

**Review by rtroper — APPROVED** [review]

**Review by bill-baumgartner — APPROVED** [review]

File: TranslatorReasonerAPI.yaml (hunk 2 above)
Comment (edeutsch) [reply TranslatorReasonerAPI.yaml:920]: What did we decide to do here? Do we allow MetaEdge.attributes to be "nullable: true" or do we switch to "default: []"?
I recall some spirited discussion, but the conclusion does not appear to be captured in the minutes.
I vaguely recall that there were more proponents of "nullable: true", but it is very hazy.

Comment (patrickkwang) [reply TranslatorReasonerAPI.yaml:920]: We want to allow both null and empty, to distinguish between "I'm not telling you" and "I'm telling you there's nothing".

**Pull request merged**